llpc/context/llpcCompiler.cpp

```diff
@@ -888,9 +890,17 @@ Result Compiler::buildPipelineWithRelocatableElf(Context *context, ArrayRef<cons
 //
 // @param [in] nodes : user data nodes
 // @param nodeCount : number of user data nodes
+#if LLPC_CLIENT_INTERFACE_MAJOR_VERSION < 41
 static bool hasUnrelocatableDescriptorNode(const ResourceMappingNode *nodes, unsigned nodeCount) {
   for (unsigned i = 0; i < nodeCount; ++i) {
     const ResourceMappingNode *node = nodes + i;
+#else
+static bool hasUnrelocatableDescriptorNode(const ResourceMappingData *resourceMapping, ShaderStage shaderStage) {
+  for (unsigned i = 0; i < resourceMapping->userDataNodeCount; ++i) {
+    if ((resourceMapping->pUserDataNodes[i].visibility & shaderStageToMask(shaderStage)) == 0)
+      continue;
+    const ResourceMappingNode *node = &resourceMapping->pUserDataNodes[i].node;
+#endif
```

Comment:
Preprocessor directives that aren't well-nested with C++ constructs are generally forbidden (some exceptions around multi-line conditionals exist).

We should really add that to the CodingStandards...